Marlin/src/lcd/extensible_ui/lib/lulzbot/ftdi_eve_lib/compat.h

```diff
@@ -197,6 +197,7 @@
   #define SERIAL_ECHO_START()
   #define SERIAL_ECHOLNPGM(str)        Serial.println(F(str))
   #define SERIAL_ECHOPGM(str)          Serial.print(F(str))
+  #define SERIAL_ECHO_MSG(str)         Serial.println(str)
```

Comment:
Actually, this mirrors `SERIAL_ECHOLN`. The `SERIAL_ECHO_MSG` macro calls `SERIAL_ECHO_START`.

Reply:
It appears that every instance of `SERIAL_ECHO_MSG` in the `lulzbot` folder is using a literal string. The Marlin version of `SERIAL_ECHO_MSG` additionally expects a PROGMEM string, so this macro could use `F()` to ensure strings used with this macro are being stored in program memory.

Reply:
@Thinkyhead: I don't understand the semantic differences between `SERIAL_ECHO_MSG` and `SERIAL_ECHOLNPGM`. I think the code originally had `SERIAL_ECHOLNPGM` but maybe you changed it to `SERIAL_ECHO_MSG`?

Anyhow, it would seem like the macro in `compat.h` is indeed wrong. It should be:

```
#define SERIAL_ECHO_MSG(str)         Serial.println(F(str))
```

Nothing in "compat.h" comes into play when compiling Marlin, but it does come into play when compiling any of the FTDI stuff outside of Marlin. My intent was to make it generic enough so it could be used in any Arduino app.

Reply:
The `SERIAL_ECHO_MSG` macro sends 'echo:' followed by `SERIAL_ECHOLNPGM`. It is used for single-line messages that should be echoed by the host.
```cpp
#define SERIAL_ECHO_MSG(S)          do{ SERIAL_ECHO_START(); SERIAL_ECHOLNPGM(S); }while(0)
```